`make haddock` fails

@knrafto, please note that `make haddock` is failing after [this](https://github.com/agda/agda/commit/e4866bd5b76993945a268dd46364a4d29a712ed6) commit:

```bash
$ make haddock
cabal v1-configure --builddir=./dist-2.6.3-ghc-9.0.1 --enable-tests -foptimise-heavily -j1 --disable-library-profiling  -fenable-cluster-counting --ghc-options="+RTS -M6G -RTS" 
cabal: unrecognized 'v1-configure' option `-j1'
Makefile:283: recipe for target 'haddock' failed
make: *** [haddock] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -122,7 +122,8 @@ STACK_INSTALL_BIN_OPTS = --no-library-profiling \
                          $(STACK_INSTALL_OPTS)
 
 CABAL_CONFIGURE_OPTS = $(SLOW_CABAL_INSTALL_OPTS) \
-                       $(CABAL_INSTALL_BIN_OPTS)
+                       --disable-library-profiling \
+                       $(CABAL_INSTALL_OPTS)
 
 ##############################################################################
 ## Installation

